src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerActor.scala

```diff
@@ -276,22 +287,25 @@ private[impl] class GroupManagerActor(
         }
       }
 
-      app.copy(
-        portDefinitions = mergeServicePortsAndPortDefinitions(app.portDefinitions, servicePorts),
-        container = newContainer.orElse(app.container)
+      newApp.copy(
+        portDefinitions = mergeServicePortsAndPortDefinitions(newApp.portDefinitions, servicePorts),
+        container = newContainer.orElse(newApp.container)
       )
     }
 
-    val dynamicApps: Set[AppDefinition] =
-      to.transitiveApps.map {
-        // assign values for service ports that the user has left "blank" (set to zero)
-        case app: AppDefinition if app.hasDynamicServicePorts => assignPorts(app)
-        case app: AppDefinition =>
-          // Always set the ports to service ports, even if we do not have dynamic ports in our port mappings
-          app.copy(
-            portDefinitions = mergeServicePortsAndPortDefinitions(app.portDefinitions, app.servicePorts)
-          )
-      }
+    val dynamicApps: Iterator[AppDefinition] =
+      to.transitiveApps
```

Comment:
I just wander how big is the difference if you would use `.view` instead of iterator :)